src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
                 {
-                    assm = Assembly.Load(new AssemblyName(assemblyName));
+                    try
```

Comment:
If the AssemblyName constructor throws, it's because the assembly name itself is badly formed. That's an exception that really shouldn't be swallowed, in my opinion.

I realize, though, that may be a "breaking change" so I won't complain too hard if you put it inside the try.

(On the similar note, catching every exception indiscriminately out of Assembly.Load() is also bound to be controversial - one more reason why I'd like to have just copy of that logic.)